src/Race.js

```diff
@@ -0,0 +1,80 @@
+import {
+  getKeyArrayHasTargetValueInMap,
+  getMapFilledZero,
+  getMaxValueInMap,
+  getRepeatedString,
+  pickNumberInRange,
+  print,
+} from './lib/utils.js';
+
+class Race {
+  static #MIN_RANDOM = 0;
+
+  static #MAX_RANDOM = 9;
+
+  static #MOVE_FORWARD_THRESHOLD = 4;
+
+  static #TRACE_CHARACTER = '-';
+
+  static #FORWARD_STEP = 1;
+
+  #carArray;
+
+  #tryCount;
+
+  #carTraceMap;
+
+  constructor(carArray, tryCount) {
+    this.#carArray = carArray;
+    this.#tryCount = tryCount;
+
+    this.#carTraceMap = getMapFilledZero(this.#carArray);
+  }
+
+  run() {
+    for (let round = 0; round < this.#tryCount; round += 1) {
+      this.#runOneRound();
+    }
+    return this.#getWinnerCarArray;
+  }
+
+  #runOneRound() {
+    this.#carArray.forEach(car => {
+      const newPosition = Race.#getNewPosition(this.#carTraceMap.get(car));
```

Comment:
이부분은 두줄로 나누는 것이 가독성이 더 좋아보여요!
```suggestion
      const prevPosition = this.#carTraceMap.get(car)
      const newPosition = Race.#getNewPosition(prevPosition);
```

Reply:
변수에 담음으로써 가독성이 좋아졌군요.. 팁 감사합니다 ☺️☺️